src/main/kotlin/com/watermelon/context/actions/LoginAction.kt

```diff
@@ -0,0 +1,96 @@
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.passwordSafe.PasswordSafe
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.ui.Messages
+import kotlinx.serialization.json.*
+import java.net.HttpURLConnection
+import java.net.URL
+
+
+class LoginAction : AnAction() {
+    private val backendUrl = "https://app.watermelontools.com"
+    private fun sendTokenToAPI(token: String): String {
+        val apiUrl = "$backendUrl/api/extension/intellijLogin"
+        val url = URL(apiUrl)
+        val connection = url.openConnection() as HttpURLConnection
+
+        try {
+            connection.requestMethod = "POST"
+            connection.doOutput = true
+            connection.setRequestProperty("Content-Type", "application/json")
+            connection.setRequestProperty("Charset", "UTF-8")
+
+            val payload = """
+            {
+                "token": "$token"
+            }
+        """.trimIndent()
+
+            connection.outputStream.write(payload.toByteArray())
+
+            val responseCode = connection.responseCode
+            return if (responseCode == HttpURLConnection.HTTP_OK) {
+                val jsonResponse = Json.parseToJsonElement(connection.inputStream.reader().readText()).jsonObject
+                val data = jsonResponse["data"]?.jsonObject
+                val id = data?.get("id")?.toString()
+                val email = data?.get("email")?.toString()
+                // Store the tokens
+                val passwordSafe = PasswordSafe.instance
+                passwordSafe.setPassword(CredentialAttributes("WatermelonContext.id"), id)
+                passwordSafe.setPassword(CredentialAttributes("WatermelonContext.email"), email)
+                connection.inputStream.reader().readText()
+            } else {
+                // Handle non-200 HTTP responses
+                "Error: $responseCode"
+            }
+        } finally {
+            connection.disconnect()
+        }
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        // Open webpage
+        BrowserUtil.browse("$backendUrl/intellij")
+
+        // Open a dialog for user input
+        val email = Messages.showPasswordDialog(
+            e.project,
+            "Please enter your email:",
+            "Email Input",
+            null
+        )
+
+        if (email.isNullOrEmpty()) {
+            Messages.showMessageDialog(
+                e.project,
+                "Email should not be empty.",
+                "Input Error",
+                Messages.getErrorIcon()
+            )
+        } else {
+            sendTokenToAPI(email)
+            Messages.showMessageDialog(
+                e.project,
+                "You have logged in to Watermelon Context as $email.",
+                "Success",
+                Messages.getInformationIcon()
+            )
+        }
+    }
+
+    override fun update(e: AnActionEvent) {
```

Comment:
Following the logic of the comment above: A TODO here is to change this validation to work not for an email, but for a hash generated key soon. But it's good for now.